addons/account/models/chart_template.py

```diff
@@ -657,6 +658,7 @@ def generate_account(self, tax_template_ref, acc_template_ref, code_digits, comp
                 account_template.tax_ids = sale_tax
             elif account_template.user_type_id.internal_group == 'expense':
                 account_template.tax_ids = purchase_tax
+            self.env['account.root'].search([('name', '=', account_template.code[:1])]) or self.env['account.root'].create({'name': account_template.code[:1]})
```

Comment:
:boom: performances.... You don't need it as your magic compute will create the root. You can also remove the `root_id` field on `account.account.template`